src/main/java/io/pinecone/model/CreateIndexRequest.java

```diff
@@ -0,0 +1,197 @@
+package io.pinecone.model;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.pinecone.PineconeValidationException;
+
+public class CreateIndexRequest {
+    private String indexName;
+
+    private Integer dimension;
+
+    private String metric = "cosine";
+
+    private Integer pods = 1;
+
+    private Integer replicas = 1;
+
+    private String podType = "p1.x1";
+
+    private IndexMetadataConfig metadataConfig;
+
+    private String sourceCollection;
+
+    public CreateIndexRequest() {
+    }
+
+    public CreateIndexRequest indexName(String name) {
+        this.indexName = name;
+        return this;
+    }
+
+    /**
+     * The name of the index to be created. The maximum length is 45 characters.
+     * @return name
+     **/
+    @javax.annotation.Nullable
```

Comment:
What's the purpose of this annotation? We don't ever want to create an index wtihout a name, so it should not be Nullable. Maybe you meant `Nonnull`?